### PR TITLE
Adding configurable file size limit for application attachments

### PR DESCRIPTION
**Pull request by viktorcsimma**
Now, it is not limited to 2 MB, but a limit can be given in `.env` (the default is 10 MB = 10 000 000 B).

This applies to language exam files and profile pictures, too. In the future, these could be separated to different `.env` entries.

And as an additional nicety, language exam files are new opened on a new tab.

### CODE REVIEW

File: app/Utils/ApplicationHandler.php [hunk 1]
```diff
@@ -55,7 +55,7 @@ public function storeQuestionsData(Request $request, User $user): void
     public function storeFile(Request $request, $user): void
     {
         $request->validate([
-            'file' => 'required|file|mimes:pdf,jpg,jpeg,png|max:2000',
+            'file' => 'required|file|mimes:pdf,jpg,jpeg,png|max:' . config('application.file_size_limit') / 1000,
```
[the remark this answers is not in the document]
Comment (viktorcsimma) [reply app/Utils/ApplicationHandler.php]: Let's settle now with 1000.

**Review by horcsinbalint — CHANGES_REQUESTED** [review]
These values are used for cases where applicants and "normal" users use these features. So naming this with something to do with "application" is quite confusing. Having a file `config/app.php` and `config/application.php` is extremely confusing.

Why is it needed to divide with 1000 every time and not configured in kilobytes?

Do we need to make this configurable? Why not just simply change the maximum size?

**Comment by viktorcsimma** [comment]
With the naming, you are right; what about putting it into the `general` config file as something like "general file size limit"?

Bytes seem to be logical to me because the `size` attribute expects bytes; those have also been used earlier for printer file size limits.

I think it would be nice to make it configurable if we could, but I don't insist on it. WDYT?

**Comment by horcsinbalint** [comment]
I'm not insisting on changing the unit of measurement, just found the proposed changes quite confusing.

On the one hand, yes, it is used in the `size` attribute. But does the `size` attribute have any effect? I think according to [the HTML5 specification](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file) the `size` attribute is ignored in cases where `<input type="file">` is used.
If you want to change the scope of this PR by implementing a general file size limit, please note that in cases where the file size limit is larger than a reverse proxy's size limit, it should return with an error code of [413](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/413) and not pass the request to the PHP interpreter.

**Comment by viktorcsimma** [comment]
I see... so should we include a comment on this limitation in .env.example?

Until then, I'll simply remove the `size` attributes; thanks for pointing that out.

**Comment by horcsinbalint** [comment]
But as the printer's file upload limit is set in bytes, I would also use bytes on this PR. So yeah, changing the units of measurements is not necessary, I think it would be out of scope of this PR. (If I were the author, I would open a separate PR for changing the unit of measurement and drop the unnecessary `size` attributes.)

**Comment by viktorcsimma** [comment]
So should we have a small, immediately mergeable PR that just raises the limits until then?

**Review by horcsinbalint — APPROVED** [review]
LGTM. We should probably open a separate issue for clearing up unnecessary `size` attributes and changing the unit of measurement to kilobytes.

**Comment by viktorcsimma** [comment]
Okay; thank you:)

**Pull request merged**